src/lib/geojson_utils.js

```diff
@@ -21,18 +23,19 @@
  *
  */
 exports.calcTraceToLineCoords = function(calcTrace) {
-    var trace = calcTrace[0].trace,
-        connectgaps = trace.connectgaps;
+    var trace = calcTrace[0].trace;
+    var connectgaps = trace.connectgaps;
 
-    var coords = [],
-        lineString = [];
+    var coords = [];
+    var lineString = [];
 
     for(var i = 0; i < calcTrace.length; i++) {
         var calcPt = calcTrace[i];
+        var lonlat = calcPt.lonlat;
 
-        lineString.push(calcPt.lonlat);
-
-        if(!connectgaps && calcPt.gapAfter && lineString.length > 0) {
+        if(lonlat[0] !== BADNUM && lonlat[1] !== BADNUM) {
```

Comment:
This is great for robustness (so this comment is nonblocking), but for 🐎  (with sufficient test coverage) we could ensure in calc that if the point is invalid, BOTH components get set to `BADNUM`, so you only need one test.